tool/uriparse.c

```diff
@@ -43,7 +43,7 @@
 #ifdef _WIN32
 # include <winsock2.h>
 # include <ws2tcpip.h>
-# ifdef __MINGW32__
+# if defined(__MINGW32__) && !defined(InetNtopA)
 WINSOCK_API_LINKAGE const char * WSAAPI inet_ntop(
 		int af, const void *src, char *dst, socklen_t size);
```

Comment:
You may want to introduce the same define mingw-w64 uses so !defined(InetNtopA) doesn't get triggered again, this is in the case that this ever potentially gets moved to a header file or somewhere else where it could be used multiple times